airmozilla/popcorn/views.py

```diff
@@ -51,11 +42,68 @@ def event_meta_data(request):
 
     assert response.status_code == 302, response.status_code
 
+    purl = urlparse(response.headers['location'])
+    location = r'{0}://{1}{2}'.format(purl.scheme, purl.hostname, purl.path)
+
+    return location
+
+
+def get_thumbnail(event):
+    image = event.picture and event.picture.file or event.placeholder_img
+    geometry = '160x90'
+    crop = 'center'
+
+    return thumbnail(image, geometry, crop=crop)
+
+
+@add_cors_header('*')
+@json_view
+def event_meta_data(request):
+    slug = request.GET.get('slug')
+    event = get_object_or_404(Event, slug=slug)
```

Comment:
One cool feature in tornado (the event driven python web framework) is that you pick things up from the request using `self.get_argument('slug')` and it automatically raises a 400 error for you if that variable is missing or an empty string. And if you don't want that to be required you can do things like: `order_by = self.get_argument('order', 'created')` which sets a default.